pkg/mark/link.go

```diff
@@ -0,0 +1,69 @@
+package mark
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/kovetskiy/mark/pkg/confluence"
+)
+
+type Link struct {
+	MDLink string
+	Link   string
+}
+
+// ResolveRelativeLinks finds links in the markdown, and replaces one pointing to other Markdowns either by own link (if not created to Confluence yet) or witn actual Confluence link
+func ResolveRelativeLinks(
+	api *confluence.API,
+	markdown []byte,
+	base string,
+) ([]Link, error) {
+	var (
+		links []Link
+	)
+	re := regexp.MustCompile("\\[.*?\\]\\((.*?)\\)")
+	submatchall := re.FindAllStringSubmatch(string(markdown), -1)
+	for _, element := range submatchall {
+		possibleMDFile := element[1]
+		filepath := filepath.Join(base, possibleMDFile)
+		if _, err := os.Stat(filepath); err == nil {
+			linkMarkdown, err := ioutil.ReadFile(filepath)
+			if err != nil {
+				continue
+			}
+			meta, _, err := ExtractMeta(linkMarkdown)
+			if err != nil {
+				continue
+			}
+			link := fmt.Sprintf("%s/display/%s/%s", api.BaseURL, meta.Space, url.QueryEscape(meta.Title))
+			confluencePage, err := api.FindPage(meta.Space, meta.Title)
```

Comment:
Don't ignore the error, report it to the user.

Reply:
Oops, you resolved the conversation but forgot to handle err returned by api.FindPage

Reply:
Hmm... there could be some print that api call failed, but at the end the default value is put in that case to the conversion so it's not fatal (same as if page is not created yet to Confluence).